addons/suggest_locations/models/stock_move_line.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+from collections import defaultdict
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+from .suggest_locations_policy import SUGGEST_LOCATION_REGISTRY
+from functools import lru_cache
+
+DROPABLE_STATES = ("assigned", "partially_available")
+CONTRAINTS_REQUIRING_A_CHECK = ("enforce", "enforce_with_empty")
+WITH_EMPTY_LOCATIONS = ("suggest_with_empty", "enforce_with_empty")
+VIEW_SET = set(["view"])
+
+
+class MoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    @lru_cache(maxsize=10)
+    def _get_policy_class(self, picking_type, preprocessing=False):
+        """Get the policy for suggesting locations"""
+        try:
+            policy_name = picking_type.u_suggest_locations_policy
+            policy_class = SUGGEST_LOCATION_REGISTRY[policy_name]
+            return policy_class(self.env, preprocessing)
+        except KeyError:
+            raise ValueError(
+                _(f"Policy with name={picking_type.u_suggest_locations_policy} could not be found")
+            )
+
+    def suggest_locations(self, picking_id=None, values=None, limit=30):
+        """Suggest locations for move line, either by self or picking_type and values"""
+        # Validate preconditions
+        if not self and (not picking_id or not values):
+            raise ValueError(
+                _(
+                    "Missing information! If suggest locations is not done via self, picking_id "
+                    + "and values must be provided!"
+                )
+            )
+
+        # Set the picking_id if not set, and the picking_type
+        if self and not picking_id:
+            picking_id = self.picking_id
+            picking_type = self.u_picking_type_id
+        elif picking_id:
+            picking_type = picking_id.picking_type_id
```

Comment:
Is this a picking id or a picking recordset? otherwise we cannot do this. Probably the parameter should be a picking, and convert picking_id into picking at `_process_suggested_inputs()`